tests/schema/tests.py

```diff
@@ -2647,6 +2648,11 @@ def test_add_field_use_effective_default(self):
             item = cursor.fetchall()[0]
             self.assertEqual(item[0], None if connection.features.interprets_empty_strings_as_nulls else '')
 
+    def test_effective_default_db_default(self):
+        """#31206 - effective_default() should be handle database defaults"""
+        with connection.schema_editor() as editor:
+            editor.effective_default(DateTimeField(default=Now()))
```

Comment:
We should test a real use case instead of checking that `effective_default()` doesn't crash, e.g. altering field with changing a default value from `timezone.now` to `Now()` (I think that will be quite common change when we document and support this API).

Reply:
People will currently need to create their own field in order to overwrite db_returning, since ticket-30032 is not merged yet. This failure is actually an edge case and only happened when someone tried to switch from `Now` to `TransactionNow`. I recreate that very use case in a test, I believe this to be the most common case for now. Is that OK?

Reply:
This test checks only if `effective_default()` crash or not, that's not something that we would like to test. Changing `DateTimeField(default=timezone.now))` to `CreatedField()` (from our test suite) creates the following migration:
```python
    operations = [
        migrations.AlterField(
            model_name='returningmodel',
            name='created',
            field=test_31206.models.CreatedField(default=django.db.models.functions.datetime.Now),
        ),
    ]
```
We can add such test to the `schema/tests.py`. If you will have any issue with adding such test I can help you tomorrow.

Reply:
Ok, I implemented a similar case now, which should cover more use cases as well.